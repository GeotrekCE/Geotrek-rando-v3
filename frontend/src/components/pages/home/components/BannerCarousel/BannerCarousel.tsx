import { LargeStyledArrow } from 'components/Carousel';
import Slider, { CustomArrowProps } from 'react-slick';
import styled, { css } from 'styled-components';
import { colorPalette, desktopOnly, getSpacing } from 'stylesheet';

interface BannerCarouselProps {
  picturesUrl: string[];
}

export const BannerCarousel: React.FC<BannerCarouselProps> = ({ picturesUrl }) => {
  return (
    <div data-testid="carousel" id="banner_carousel">
      <Slider
        className="h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
        infinite
        nextArrow={<NextArrow />}
        prevArrow={<PrevArrow />}
        swipe={false}
        autoplay
        speed={1000}
        autoplaySpeed={5000}
        pauseOnHover
      >
        {picturesUrl.map((pictureUrl, i) => (
          <img
            loading={i === 0 ? 'eager' : 'lazy'}
            src={pictureUrl}
            key={i}
            className="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
            alt=""
          />
        ))}
      </Slider>
    </div>
  );
};

const PrevArrow = (props: CustomArrowProps) => {
  const { className = '', onClick } = props;
  return <StyledLeftArrow className={className} onClick={onClick} type="button" />;
};

const NextArrow = (props: CustomArrowProps) => {
  const { className = '', onClick } = props;
  return <StyledRightArrow className={className} onClick={onClick} type="button" />;
};

const BannerStyledArrow = styled(LargeStyledArrow)`
  padding: ${getSpacing(20)} ${getSpacing(5)};
  opacity: 1;
  &::before {
    text-shadow: 0 0 20px ${colorPalette.home.shadowOnImages};
    ${desktopOnly(
      css`
        font-size: 30px;
      `,
    )}
  }
  ${desktopOnly(
    css`
      padding: ${getSpacing(40)} ${getSpacing(12)};
    `,
  )}
`;

const StyledRightArrow = styled(BannerStyledArrow)`
  right: 0;
`;

const StyledLeftArrow = styled(BannerStyledArrow)`
  left: 0;
`;
