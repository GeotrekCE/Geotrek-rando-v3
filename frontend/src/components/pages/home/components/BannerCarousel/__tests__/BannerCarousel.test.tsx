import { render } from 'services/testing/reactTestingLibraryWrapper';
import { BannerCarousel } from '../BannerCarousel';
describe('BannerCarousel', () => {
  it('should render correctly', () => {
    const component = render(<BannerCarousel picturesUrl={['photoA.jpg', 'photoB.jpg']} />);

    expect(component).toMatchSnapshot();
  });
});
