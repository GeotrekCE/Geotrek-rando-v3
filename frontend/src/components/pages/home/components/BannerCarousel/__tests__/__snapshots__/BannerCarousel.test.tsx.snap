// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCarousel should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="carousel"
        id="banner_carousel"
      >
        <div
          class="slick-slider h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full slick-initialized"
          dir="ltr"
        >
          <div
            class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledLeftArrow-sc-q9y4d7-2 hBtEpW fuiwDG nGhjU MQxHa slick-arrow slick-prev"
          />
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="-1"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="photoB.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="photoA.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="photoB.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="2"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="photoA.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="3"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="photoB.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledRightArrow-sc-q9y4d7-1 hBtEpW fuiwDG nGhjU gSQeFB slick-arrow slick-next"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="carousel"
      id="banner_carousel"
    >
      <div
        class="slick-slider h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full slick-initialized"
        dir="ltr"
      >
        <div
          class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledLeftArrow-sc-q9y4d7-2 hBtEpW fuiwDG nGhjU MQxHa slick-arrow slick-prev"
        />
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt=""
                  class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                  src="photoB.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt=""
                  class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                  src="photoA.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt=""
                  class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                  src="photoB.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt=""
                  class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                  src="photoA.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="3"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt=""
                  class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                  src="photoB.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledRightArrow-sc-q9y4d7-1 hBtEpW fuiwDG nGhjU gSQeFB slick-arrow slick-next"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
